src/covariances.jl

```diff
@@ -0,0 +1,87 @@
+# ------------------------------------------------------------------
+# Licensed under the MIT License. See LICENCE in the project root.
+# ------------------------------------------------------------------
+
+"""
+    design_matrix(comps)
+    
+Converts a vector of Composition objects into a Matrix of numbers.
+"""
+function design_matrix(comps)
+    matrix = reduce(hcat, components.(comps)[:,1])'
+    return matrix
+end
+
+"""
+    get_subcomposition_matrix(matrix, entries)
+
+Returns the subcomposition matrix given the `entries` (selected columns) from the `matrix`.
+"""
+function get_subcomposition_matrix(matrix, entries)
+    subcomposition_matrix = matrix[:, entries]
+    return subcomposition_matrix
+end
+
+"""
+    comp_var_array(comps)
+
+Returns the compositional variation array, from definition 4.3 of Aitchson - The Statistical Analysis of Compositional Data.
+"""
+function comp_var_array(comps)
+    dm = design_matrix(comps)
+    N, D = size(dm)
+    cva = zeros(D, D)
+
+    for i in 1:D
+        for j in i+1:D
+            log_ratios = log.(dm[:, i] ./ dm[:, j])
+            cva[j, i] = sum(log_ratios) / N
+            cva[i, j] = sum((log_ratios .- cva[j, i]).^2 ) / N
+        end
+    end
+
+    return cva
+end
+
+"""
+    variation_matrix(comps)
+
+Return the variation matrix, definition 4.4 of Aitchson - The Statistical Analysis of Compositional Data.
+"""
+function variation_matrix(comps)
+    dm = design_matrix(comps)
+    N, D = size(dm)
+    va = zeros(D, D)
```

Comment:
Again, maybe we don't need to allocate a full matrix in the beginning to fill in the values of the covariances. Let's discuss in the meeting today.